src/pkg/services/m365/api/graph/metadata/metadata_test.go

```diff
@@ -152,3 +152,108 @@ func (suite *MetadataUnitSuite) TestIsMetadataFile_Directories() {
 		}
 	}
 }
+
+func (suite *MetadataUnitSuite) TestIsMetadataFile() {
+	table := []struct {
+		name       string
+		service    path.ServiceType
+		category   path.CategoryType
+		isMetaFile bool
+		expected   bool
+	}{
+		{
+			name:     "onedrive .data file",
+			service:  path.OneDriveService,
+			category: path.FilesCategory,
+		},
+		{
+			name:     "sharepoint library .data file",
+			service:  path.SharePointService,
+			category: path.LibrariesCategory,
+		},
+		{
+			name:     "group library .data file",
+			service:  path.GroupsService,
+			category: path.LibrariesCategory,
+		},
+		{
+			name:     "group conversations .data file",
+			service:  path.GroupsService,
+			category: path.ConversationPostsCategory,
+		},
+		{
+			name:       "onedrive .meta file",
+			service:    path.OneDriveService,
+			category:   path.FilesCategory,
+			isMetaFile: true,
+			expected:   true,
+		},
+		{
+			name:       "sharepoint library .meta file",
+			service:    path.SharePointService,
+			category:   path.LibrariesCategory,
+			isMetaFile: true,
+			expected:   true,
+		},
+		{
+			name:       "group library .meta file",
+			service:    path.GroupsService,
+			category:   path.LibrariesCategory,
+			isMetaFile: true,
+			expected:   true,
+		},
+		{
+			name:       "group conversations .meta file",
+			service:    path.GroupsService,
+			category:   path.ConversationPostsCategory,
+			isMetaFile: true,
+			expected:   true,
+		},
+		// For services which don't have metadata files, make sure the function
+		// returns false. We don't want .meta suffix (assuming it exists) in
+		// these cases to be interpreted as metadata files.
+		{
+			name:       "exchange service",
+			service:    path.ExchangeService,
+			category:   path.EmailCategory,
+			isMetaFile: true,
+		},
+		{
+			name:       "group channels",
+			service:    path.GroupsService,
+			category:   path.ChannelMessagesCategory,
+			isMetaFile: true,
+		},
+		{
+			name:       "lists",
+			service:    path.SharePointService,
+			category:   path.ListsCategory,
+			isMetaFile: true,
+		},
+	}
+
+	for _, test := range table {
```

Comment:
I'm wondering if a bit of refactoring could give better test coverage while maybe making things a little more clear. I was thinking of a few things:
1. define a test input struct outside of the test so it can be used in multiple functions
2. create a `runTest` function (maybe with a better name) that has (most of) the logic that's currently in the for loop
3. create two `[]testInput` variables: one for service/categories that support metadata and another for ones that don't
4. create two `TestX` functions: one for the metadata suffix and one for the data suffix (alternatively: create three `TestX` functions for has metadata, doesn't have metadata, and data)
5. call into the `runTest` function in the `TestX` functions

The test struct could be similar to
```go
type testInput struct {
  service path.ServiceType
  category path.CategoryType
  fileName string // <- should include .data/.meta suffix
  expected bool
}
```

if you define the metadata results as two slices in the file (not a function) then the data suffix test can loop over those and just build the correct input struct like below
```go
func TestDataSuffix() {
  allTests := []testInput{}

  // Need to append to an empty slice so it doesn't modify the first list passed.
  for _, input := range append(append([]testInput{}, isMetaTests...), isNotMetaTests...) {
    allTests = append(allTests, testInput{
      service: input.service,
      category: input.category,
      fileName: "testfile" + dataSuffix,
    })
  }

  // Can switch up this logic to better fit your preference.
  for _, test := range allTests {
    suite.Run(test.service.HumanString() + test.category.HumanString() + "Data", func() {
      runTest(test)
    })
  }
}
```

The above has the advantage of running tests for both meta and data suffixes for every service/category. It also only requires a service/category be defined in a single place: the list that determines if it has a metadata file or not